pkg/kgo/source.go

```diff
@@ -166,7 +160,10 @@ type ProcessFetchPartitionOptions struct {
 	Partition int32
 
 	// recordsPool is for internal use only.
-	recordPool recordsPool
+	recordPool *recordsPool
+
+	// decompressBufferPool is for internal use only.
+	decompressBufferPool *pool.BucketedPool[byte]
```

Comment:
my suggestion was to have this exported, since we want to use it in the fetchers implementation in Mimir. 

(For consistency maybe also recordPool can be exported? So far we don't need this in Mimir, so i leave it up to you)

Reply:
_addressed @ 47adbd15302565bc4e2c3353949f22304bbd1aa9_